### ISSUE TITLE
Concurrent execution of master_apply_delete_command fails

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#1396]
### Setup
Range partition table t1(a int, b int) has 2 shards.
Shard 1 : 1 - 1000
Shard 2 : 1001 - 2000

### Connection 1 : 
```sql
begin;
select * from master_apply_delete_command('delete from t2 where a < 1001');
commit;
```

### Connection 2 : 
```sql
begin;
select * from master_apply_delete_command('delete from t2 where a < 2001');
commit;
```

### STEPS
- Connection 1 goes first, issues delete command but does not commit, 
- Connection 2 issues command, gets blocked, waits for connection 1 to commit
- Connection 1 commits, connection 2 is unblocked
- Connection 2 fails since it can not find shard1,  it does not delete shard 2 either

This does not cause any corruption or data loss, also user is notified of failure. Re-running the command on connection 2 works fine. We may live with it.

Note : this is similar to #1381 and likely to be fixed by #1382

**Comment by metdos** [comment]
Closing in favor of #727.
